lib/redis/manager.js

```diff
@@ -61,18 +61,19 @@ class StateManager {
   reset () {
     this._versions = {'*': {'*': -1}};
     const scan = (node, cursor) => {
+      let newCursor = 0;
+
       return node.scan(cursor, 'MATCH', 'cluster*', 'COUNT', 1000)
         .then(response => {
-          const [newCursor, keys] = response;
-
-          return Bluebird.all(keys
-            .filter(key => key !== 'cluster:discovery')
-            .map(key => node.del(key)))
-            .then(() => {
-              if (parseInt(newCursor) > 0) {
-                return scan(node, newCursor);
-              }
-            });
+          const keys = response.keys.filter(key => key !== 'cluster:discovery');
+          newCursor = response.newCursor;
+
+          return Bluebird.map(keys, key => node.del(key));
+        })
+        .then(() => {
+          if (parseInt(newCursor) > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add the base to this "parseInt" call. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2427)